src/tree.ts

```diff
@@ -307,23 +307,74 @@ export class Tree {
   }
 
   /**
+   * Set a current folding type: expanded, collapsed or leaf.
+   */
+  private _setFoldingType(): void {
+    if (this.childrenShouldBeLoaded()) {
+      this.node._foldingType = FoldingType.Collapsed;
+    } else if (this._children) {
+      this.node._foldingType = FoldingType.Expanded;
+    } else {
+      this.node._foldingType = FoldingType.Leaf;
+    }
+  }
+
+  /**
    * Get a current folding type: expanded, collapsed or leaf.
    * @returns {FoldingType} A folding type of the current tree.
    */
   public get foldingType(): FoldingType {
     if (!this.node._foldingType) {
-      if (this.childrenShouldBeLoaded()) {
-        this.node._foldingType = FoldingType.Collapsed;
-      } else if (this._children) {
-        this.node._foldingType = FoldingType.Expanded;
-      } else {
-        this.node._foldingType = FoldingType.Leaf;
-      }
+      this._setFoldingType();
     }
     return this.node._foldingType;
   }
 
   /**
+   * Get a css class for element which displayes folding state - expanded, collapsed or leaf
+   * @returns {string} A string icontaining css class (classes)
+   */
+  public get foldingCssClass(): string {
+    if (!this.node._foldingType) {
```

Comment:
It should be like this
```
public get foldingCssClass(): string {
    return this.getCssClassesFromSettings() || this.foldingType.cssClass;
  }

  private getCssClassesFromSettings(): string {
    if (!this.node._foldingType) { 
      this._setFoldingType();
    } 
    
    if (this.node._foldingType === FoldingType.Collapsed) {
       return _.get(this.node.settings, 'cssClasses.collapsed', null);
    } else if (this.node._foldingType === FoldingType.Expanded) {
       return _.get(this.node.settings, 'cssClasses.expanded', null);
    }
    
    return _.get(this.node.settings, 'cssClasses.leaf', null);
  }
```
because when you get css classes by calling foldingCssClass, it first calles getCssClassesFromSettings which tries to access `this.node._foldingType` which hasn't have any value and you get an error - `Expression has changed after it was checked. Previous value: 'fa'. Current value: 'fa fa-caret-down'.`